core/toolbox/toolbox.js

```diff
@@ -713,13 +704,44 @@ Blockly.Toolbox.prototype.shouldDeSelectItem_ = function(oldItem, newItem) {
  *     toolbox item.
  * @param {?Blockly.SelectableToolboxItem} newItem The newly selected toolbox
  *     item.
- * @return {boolean} True if the new item should be deselected, false otherwise.
+ * @return {boolean} True if the new item should be selected, false otherwise.
  * @protected
  */
 Blockly.Toolbox.prototype.shouldSelectItem_ = function(oldItem, newItem) {
+  // Select the new item unless the old item equals the new item.
   return !!(newItem && newItem != oldItem);
 };
 
+/**
+ * Deselects the given item, marks it as unselected, and updates aria state.
+ * @param {!Blockly.SelectableToolboxItem} item The previously selected
+ *     toolbox item which should be deselected.
+ * @protected
+ */
+Blockly.Toolbox.prototype.deselectItem_ = function(item) {
+  this.selectedItem_ = null;
+  this.previouslySelectedItem_ = item;
+  item.setSelected(false);
+  Blockly.utils.aria.setState(/** @type {!Element} */ (this.contentsDiv_),
+      Blockly.utils.aria.State.ACTIVEDESCENDANT, '');
+};
+
+/**
+ * Selects the given item, marks it selected, and updates aria state.
+ * @param {?Blockly.SelectableToolboxItem} oldItem The previously selected
+ *     toolbox item.
+ * @param {!Blockly.SelectableToolboxItem} newItem The newly selected toolbox
```

Comment:
newItem can be null when someone wants to clear the selection.

Reply:
It could be null in shouldSelectItem but then that would return false and we would never get to this code. If it could be null there would be a bug in the prior existing code because we call newItem.setSelected(true) which would error if it were null. And it just doesn't make sense to set null as selected, if they just want to clear the selection then only deselectItem_ should be called. So nothing happening in this method should actually happen if newItem is null, so I think it's fine to require it require non-null since we already have shouldSelectItem to prevent it from being called with null.

Reply:
Ok. That makes sense. I think to fix travis then we need to cast before passing to selectItem_ since shouldSelectItem checks that the newItem is not null

Reply:
ok, kind of disappointed the compiler doesn't already know they can't be null here, but I added an explicit check for null and now it does know.